ToDoList/test/ToDoList.API.Tests/Controllers/ToDosControllerTests.cs

```diff
@@ -0,0 +1,126 @@
+    {
+        private IToDoRepository _toDoRepositorySubstitute;
+        private IUrlLocationService _urlLocationServiceSubstitute;
+        private ToDosController _controller;
+
+        private static readonly IEnumerable<ToDo> _toDoList = new[]
+        {
+            new ToDo {Id = Guid.Parse("790e8b03-aaea-46dd-9d9b-c33f3ff04090"), Text = "Dummy To Do 1"},
+            new ToDo {Id = Guid.Parse("954eccc5-2047-4dda-bcb0-e1d8d176959d"), Text = "Dummy To Do 2"},
+            new ToDo {Id = Guid.Parse("1d710f5d-4bbe-4654-906e-6c708e2bc410"), Text = "Dummy To Do 3"}
+        };
+
+        [SetUp]
+        public void SetUp()
+        {
+            _toDoRepositorySubstitute = Substitute.For<IToDoRepository>();
+            _urlLocationServiceSubstitute = Substitute.For<IUrlLocationService>();
+
+            _controller = new ToDosController(_toDoRepositorySubstitute, _urlLocationServiceSubstitute)
+            {
+                Request = new HttpRequestMessage(),
+                Configuration = new HttpConfiguration()
+            };
+        }
+
+        [Test]
+        public async Task GetToDosAsync_AllToDosReturned()
+        {
+            // Arrange
+            _toDoRepositorySubstitute.GetToDosAsync().Returns(_toDoList);
+
+            // Act
+            var response = await _controller.ExecuteAction(controller => controller.GetToDosAsync());
+            response.TryGetContentValue(out IEnumerable<ToDo> result);
+
+            // Assert
+            Assert.That(response.StatusCode, Is.EqualTo(HttpStatusCode.OK), $"Expecting status code OK, but was {response.StatusCode}");
+            Assert.That(result, Is.EqualTo(_toDoList).UsingToDoComparer(), "Todos are not equal");
+        }
+
+        [Test]
+        public async Task GetToDoAsync_CorrectToDoReturned()
+        {
+            // Arrange
+            const int itemIndex = 0;
+            var expectedToDo = _toDoList.ElementAt(itemIndex);
+            _toDoRepositorySubstitute.GetToDoAsync(expectedToDo.Id).Returns(expectedToDo);
+
+            // Act
+            var response = await _controller.ExecuteAction(controller => controller.GetToDoAsync(expectedToDo.Id));
+            response.TryGetContentValue(out ToDo result);
+
+            // Assert
+            Assert.That(response.StatusCode, Is.EqualTo(HttpStatusCode.OK), $"Expecting status code OK, but was {response.StatusCode}");
+            Assert.That(result, Is.EqualTo(expectedToDo).UsingToDoComparer(), $"{result} is not equal to expected {expectedToDo}");
+        }
+
+        [Test]
+        public async Task PostToDoAsync_IsAdded_NewToDoWithEqualRouteReturned()
+        {
+            // Arrange
+            const int itemIndex = 2;
+            var expectedToDo = _toDoList.ElementAt(itemIndex);
+
+            _toDoRepositorySubstitute.AddToDoAsync(expectedToDo).Returns(expectedToDo);
+            _urlLocationServiceSubstitute.GetNewResourceLocation(expectedToDo.Id).Returns($"todos/{expectedToDo.Id}");
```

Comment:
Consider moving the `string` into the arrange part - it's 50:50 since the string repeats twice, but constant mustn't add to readability…